internal/encryption/default_registry.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package encryption
+
+import (
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider/pbkdf2"
+	"github.com/opentofu/opentofu/internal/encryption/method/aesgcm"
+	"github.com/opentofu/opentofu/internal/encryption/registry/lockingencryptionregistry"
+)
+
+var DefaultRegistry = lockingencryptionregistry.New()
+
+func init() {
+	if err := DefaultRegistry.RegisterKeyProvider(pbkdf2.New()); err != nil {
+		panic(err)
+	}
+	if err := DefaultRegistry.RegisterMethod(aesgcm.New()); err != nil {
+		panic(err)
+	}
+}
```

Comment:
Love this new default registry :D @cam72cam you may have to rebase your AWS-KMS PR on this to include it, let's not forget =D

Reply:
I'm planning on having this merged first as it is in a bit more of a mature state.